share/smack/prelude.py

```diff
@@ -0,0 +1,290 @@
+    name = base_name + '.' + bv(bit_width),
+    attributes = [Attribute(attribute = 'inline')],
+    arguments = [Argument(name = 'i1', type = bv(bit_width)), Argument(name = 'i2', type = bv(bit_width))],
+    return_type = bv(bit_width),
+    body = 'if ' + base_name_2 + '.' + bv(bit_width) + '.bool(i1, i2) then i1 else i2'
+  )
+
+def inline_bvbuiltin_binary_pred(bit_width, base_name):
+  return Function(
+    name = base_name + '.' + bv(bit_width),
+    attributes = [Attribute(attribute = 'inline')],
+    arguments = [Argument(name = 'i1', type = bv(bit_width)), Argument(name = 'i2', type = bv(bit_width))],
+    return_type = 'bv1',
+    body = 'if ' + base_name + '.' + bv(bit_width) + '.bool(i1, i2) then 1bv1 else 0bv1'
+  )
+
+def inline_conversion(bit_width, bit_width_2, base_name, body):
+  return Function(
+    name = base_name + '.' + int(bit_width) + '.' + int(bit_width_2),
+    attributes = [Attribute(attribute = 'inline')],
+    arguments = [Argument(name = 'i', type = int(bit_width))],
+    return_type = int(bit_width_2),
+    body = body
+  )
+
+def trunc_op(bit_width, bit_width_2):
+  return Function(
+    name = '$trunc.' + int(bit_width) + '.' + int(bit_width_2),
+    attributes = [Attribute(attribute = 'inline')],
+    arguments = [Argument(name = 'i', type = int(bit_width))],
+    return_type = int(bit_width_2),
+    body = 'i'
+  )
+
+def trunc_bv_op(bit_width, bit_width_2):
+  return Function(
+    name = '$trunc.' + bv(bit_width) + '.' + bv(bit_width_2),
+    attributes = [Attribute(attribute = 'inline')],
+    arguments = [Argument(name = 'i', type = bv(bit_width))],
+    return_type = bv(bit_width_2),
+    body = 'i[' + str(bit_width_2) + ':0]'
+  )
+
+sizes = [1, 8, 16, 24, 32, 40, 48, 56, 64, 88, 96, 128]
+
+binary_ops = ['add', 'sub', 'mul', 'sdiv', 'smod', 'srem', 'udiv', 'urem', 'shl', 'lshr', 'ashr', 'and', 'or', 'xor', 'nand']
+comp_ops = ['ule', 'ult', 'uge', 'ugt', 'sle', 'slt', 'sge', 'sgt']
+eq_ops = ['eq', 'ne']
+
+binary_ops_body = ['i1 + i2', 'i1 - i2', 'i1 * i2']
+comp_ops_body = ['i1 <= i2', 'i1 < i2', 'i1 >= i2', 'i1 > i2', 'i1 <= i2', 'i1 < i2', 'i1 >= i2', 'i1 > i2']
+eq_ops_body = ['i1 == i2', 'i1 != i2']
+
+def get_prelude():
+  bpl = []
+  for i, op in enumerate(binary_ops):
+    bpl += map(lambda size: bvbuiltin_binary_op(bit_width = size, builtin_name = 'bv' + op, base_name = '$' + op).toStr(), sizes)
+    if 0 <= i <= 2:
+      bpl += map(lambda size: inline_binary_op(bit_width = size, base_name = '$' + op, body = binary_ops_body[i]).toStr(), sizes)
+    elif 3 <= i <= 7:
+      bpl += map(lambda size: builtin_binary_op(bit_width = size, builtin_name = op[1:], base_name = '$' + op).toStr(), sizes)
+    else:
+      bpl += map(lambda size: uninterpreted_binary_op(bit_width = size, base_name = '$' + op).toStr(), sizes)
+
+  for i, op in enumerate(comp_ops):
+    bpl += map(lambda size: bvbuiltin_binary_comp(bit_width = size, builtin_name = 'bv' + op, base_name = '$' + op).toStr(), sizes)
```

Comment:
Does it matter the order in which these declarations are listed? Because this could be shortened significantly:
```Python
for i, op in enumerate(comp_ops):
    for size in sizes:
        bpl += bvbuiltin_binary_cop(…)
        bpl += inline_bvbuiltin_binary_pred(…)
        …
```